pkg/controllers/jobset_controller.go

```diff
@@ -556,6 +569,56 @@ func (r *JobSetReconciler) reconcileReplicatedJobs(ctx context.Context, js *jobs
 	return nil
 }
 
+func (r *JobSetReconciler) recordJobCreationFailed(js *jobset.JobSet, err error) {
+	key := types.NamespacedName{Namespace: js.GetNamespace(), Name: js.GetName()}.String()
+	r.recordThrottledEvent(key, js, js.Spec.ReplicatedJobs, corev1.EventTypeWarning, constants.JobCreationFailedReason, err.Error())
```

Comment:
Passing `js.Spec.ReplicatedJobs` as `opaque` basically means the event is throttled as long as the object remains the same. So changing the spec causes the event to be emitted immediately.